File: src/main/java/uk/gov/justice/digital/hmpps/riskprofiler/model/PrisonSupported.java
package uk.gov.justice.digital.hmpps.riskprofiler.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.validator.constraints.Length;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import javax.persistence.Entity;
import javax.persistence.EntityListeners;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.Id;
import javax.persistence.Table;
import javax.validation.constraints.NotNull;
import java.time.LocalDateTime;

@Entity
@Table(name = "PRISON_SUPPORTED")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder(toBuilder = true)
public class PrisonSupported {
    @Id
    @Length(max = 6)
    private String prisonId;

    @NotNull
    private LocalDateTime startDateTime;
}

File: src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/model/PrisonSupported.kt
package uk.gov.justice.digital.hmpps.riskprofiler.model

import org.hibernate.validator.constraints.Length
import java.time.LocalDateTime
import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.Table
import javax.validation.constraints.NotNull

@Entity
@Table(name = "PRISON_SUPPORTED")
data class PrisonSupported(
  @Id
  @Length(max = 6)
  val prisonId: String,

  val startDateTime: @NotNull LocalDateTime? = null,
)
